src/platform/intel/cavs/include/cavs/lib/memory.h

```diff
@@ -74,6 +74,12 @@
 #define is_uncached(address) \
 	(((uint32_t)(address) & SRAM_ALIAS_MASK) == SRAM_ALIAS_BASE)
 
+static inline void *platform_shared_get(void *ptr, int bytes)
+{
+	dcache_invalidate_region(ptr, bytes);
+	return cache_to_uncache(ptr);
```

Comment:
Does the get() here imply ownership of the region ? i.e. do we have to put() it after use ?

Reply:
@lgirdwood Yes and no. get() is used only once right after allocation of shared memory region to imply the multicore access to the memory. Platforms with uncached memory can do something like code above, but other SMP platforms can just invalidate the memory without ptr transformation (@paulstelian97 FYI).
In the very near future I plan on introducing platform_shared_put or rather platform_shared_commit (per @mmaka1 suggestion) function/macro, which will be used after every read/write of shared memory to assure data synchronization between cores. Such method will do nothing on "uncached platforms", but should execute dcache wtb and invalidate on others. With coming patches it should become more clear.

Reply:
ok, doxygen comments would be good. Thanks.

Reply:
@lgirdwood Done.